### PR TITLE
#159022587 Admin user add office floor section

### PR DESCRIPTION
**Pull request by kmazi**
### What does this PR do?
- It allows the admin user to add office section

### Description of Task to be completed?
- Add floor section
- Write tests for office floor section
- Write tests for office floor section endpoints

### How should this be manually tested?
- Test out the `office-sections` endpoint with admin right

### Screenshot
- 
screen shot 2018-08-01 at 4 00 30 am

### Relevant Stories
- #159022587

### CODE REVIEW

File: core/models/officeblock.py [hunk 1]
```diff
@@ -34,3 +34,15 @@ class Meta:
 
     def __str__(self):
         return "{}".format(self.number)
+
+
+class FloorSection(models.Model):
```
Comment (collinmutembei) [inline core/models/officeblock.py]: Rename the model to `OfficeFloorSection` and the fields as follows:
- `section_name` to `name`.
- `floor_number` to `floor`.

File: core/models/officeblock.py [hunk 2]
```diff
@@ -34,3 +34,15 @@ class Meta:
...
+    section_name = models.CharField(max_length=100, unique=True)
+    floor_number = models.ForeignKey(OfficeFloor, on_delete=models.PROTECT)
+
+    class Meta:
```
Comment (collinmutembei) [inline core/models/officeblock.py]: Include `verbose_name` in class Meta.

File: core/migrations/0018_floorsection.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Generated by Django 2.0.1 on 2018-07-29 22:03
```
Comment (collinmutembei) [inline core/migrations/0018_floorsection.py]: After the changes to the model, ensure you check in only one migration file reflecting the changes.

File: core/models/officeblock.py (hunk 1 above)
Comment (collinmutembei) [reply core/models/officeblock.py]: Refactor all references to the model to reflect the rename.

File: core/tests/test_floor_sections.model.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+from rest_framework.exceptions import ValidationError
+
+from ..models import OfficeBlock, FloorSection
+
+from core.tests import CoreBaseTestCase
+
+User = get_user_model()
+
+
+class OfficeBlockModelTest(CoreBaseTestCase):
```
Comment (collinmutembei) [inline core/tests/test_floor_sections.model.py]: To keep the tests DRY, move the test cases specific to an office section to `core/tests/test_office.py`.

**Comment by collinmutembei** [comment]
@kmazi Peruse through the [best-practices](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices) repo and kindly refrain from going out of the recommended conventions. Update the PR title and description accordingly.

**Comment by QUDUSKUNLE** [comment]
@collinmutembei I've worked on the changes requested on this PR. @kmazi is no longer on the team.

File: core/tests/test_floor_sections.model.py (hunk 4 above)
Comment (collinmutembei) [reply core/tests/test_floor_sections.model.py]: Move the tests cases to `core/tests/test_office.py` and delete this module. Also, refactor the `test_office.py` module to keep it DRY. Case in point; `setUp` has some repetitive statements.
https://github.com/AndelaOSP/art-backend/blob/bbf6724bc1c7c08e31113d6921852c834c4d8e34/core/tests/test_office.py#L17-L20

https://github.com/AndelaOSP/art-backend/blob/bbf6724bc1c7c08e31113d6921852c834c4d8e34/core/tests/test_office.py#L75-L78

**Comment by QUDUSKUNLE** [comment]
@collinmutembei Changes effected. Thanks

File: core/tests/test_office.py [hunk 5]
```diff
@@ -48,20 +72,22 @@ def test_cannot_add_existing_office_block(self):
...
-        self.assertEqual(self.floor_number_counts.count(), 1)
-        new_office_floor = OfficeFloor(
-            number=10,
-            block=self.office_block)
-        new_office_floor.save()
-        self.assertEqual(self.floor_number_counts.count(), 2)
+    def test_office_floor_section_model_string_representation(self):
+        self.assertEqual(str(self.office_floor_section), "Right Wing")
+
+    def test_can_add_existing_office_floor_section(self):
```
Comment (collinmutembei) [inline core/tests/test_office.py]: Isn't this a repetition. Did you mean to test that a duplicate office floor cannot be created?
https://github.com/AndelaOSP/art-backend/blob/36232fb08b4e1d979f5c1f65e25c0b9282f1b4cb/core/tests/test_office.py#L57-L64

Comment (collinmutembei) [reply core/tests/test_office.py]: @QUDUSKUNLE Update the last test case to check for expected behaviour when there's a duplicate entry on the **OfficeFloorSection** model

Comment (QUDUSKUNLE) [reply core/tests/test_office.py]: It is not a repetition. To test a duplicate `office_section` on the same floor.

Comment (collinmutembei) [reply core/tests/test_office.py]: From what I can tell; `test_add_new_office_floor_section()` is checking for the creation behaviour on the OfficeFloorSection model, with the `floor` FK pointing to `self.office_floor`. Same thing for `test_can_add_existing_office_floor_section()`. Bear in mind that `OfficeFloorSection.objects.create()` and `OfficeFloorSection()` both create an object based-off the **OfficeFloorSection** model.

Comment (QUDUSKUNLE) [reply core/tests/test_office.py]: Yes, pointing to the same floor but different `floor_section`. Initially, it was `Right Wing` but now `Left Wing`.

Comment (collinmutembei) [reply core/tests/test_office.py]: Both **OfficeFloorSection**s are named "Left Wing".
https://github.com/AndelaOSP/art-backend/blob/36232fb08b4e1d979f5c1f65e25c0b9282f1b4cb/core/tests/test_office.py#L61
https://github.com/AndelaOSP/art-backend/blob/36232fb08b4e1d979f5c1f65e25c0b9282f1b4cb/core/tests/test_office.py#L90

File: core/models/officeblock.py [hunk 6]
```diff
@@ -34,3 +34,29 @@ class Meta:
...
+        """
+        try:
+            self.full_clean()
+        except Exception as e:
+            raise ValidationError(e)
+        super().save(*args, **kwargs)
+
+    class Meta:
+        unique_together = (('floor', 'name'),)
+        verbose_name = 'Floor Section'
```
Comment (collinmutembei) [inline core/models/officeblock.py]: Update `verbose_name` to 'Office Floor Section'

File: core/migrations/0018_auto_20180731_1400.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+                    auto_created=True,
+                    primary_key=True,
+                    serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=100)),
+                ('floor', models.ForeignKey(
+                    on_delete=django.db.models.deletion.PROTECT,
+                    to='core.OfficeFloor')),
+            ],
+            options={
+                'verbose_name': 'Floor Section',
```
Comment (collinmutembei) [inline core/migrations/0018_auto_20180731_1400.py]: The change of the verbose_name requires generating a new migration file for the name change to reflect. I recommend, you delete this migration file and generate a new one.

Comment (collinmutembei) [reply core/migrations/0018_auto_20180731_1400.py]: Also, I have noticed you are force pushing your changes...could you just create new commits instead, there's no harm in having many commits.

**Pull request merged**